R/helpers.R

```diff
@@ -0,0 +1,459 @@
+#' Helper function which calculates p-value via chi-square or fisher
+#'
+#' @param df <`tbl_df`> Dataframe that has variable and treatment columns of interest
+#' @param var <`character(1)`> Name of variable column
+#' @param treatment <`character(1)`> Name of treatment column
+#'
+#' @examples
+#'
+#' p_chi_fisher(mtcars, var = "am", treatment = "vs")
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_chi_fisher <- function(df, var, treatment) {
+
+  chisq_wrapper <- function(var, df, treatment) {
+    table <- stats::chisq.test(as.factor(df[[var]]), as.factor(df[[treatment]]))
+    return(table$p.value)
+  }
+
+  fisher_wrapper <- function(var, df, treatment) {
+    table <- stats::fisher.test(as.factor(df[[var]]), as.factor(df[[treatment]]), simulate.p.value = TRUE)
+    return(table$p.value)
+  }
+
+  chisq_wrapper <- purrr::quietly(chisq_wrapper)
+  chisq <- chisq_wrapper(var, df, treatment)
+
+  if (length(chisq$warnings) == 0) {
+    return(chisq$result)
+  } else {
+    return(fisher_wrapper(var, df, treatment))
+  }
+
+}
+
+#' Helper function which calculates p-value via t-test
+#'
+#' @inheritParams p_chi_fisher
+#'
+#' @examples
+#'
+#' p_t_test(mtcars, var = "mpg", treatment = "vs")
+#'
+#' @return <`numeric(1)`> p-value
+#'
+#' @noRd
+p_t_test <- function(df, var, treatment) {
+
+  df %>%
+    pull(var) %>%
+    split(df %>% pull(treatment)) %>%
+    stats::setNames(c("x", "y")) %>%
+    do.call(stats::t.test, .) %>%
+    `[[`("p.value")
```

Comment:
Since we're using `tidyverse` elsewhere, would it make sense to be consistent and use it here as well?  Could`set_names()`, `map()`, and `pluck()` work here?

Reply:
Yes, this definitely needs a re-write, I have no idea why this was written this way. 

I always forget that `pluck` exists, will be useful other places.

Reply:
This is doing a single t-test where x and y are the vectors to compare. Is there a cleaner way to do the `do.call` line?

One alternative is this: `{t.test(x = .$x, y = .$y)}`

Reply:
I'd use the formula syntax like you did for ANOVA.

```
form <- stats::as.formula(paste0(var, " ~ ", treatment))
stats::t.test(form, df) %>% pluck("p.value")
```